sdk/servicebus/azure-messaging-servicebus-track1-perf/src/main/java/com/microsoft/azure/messaging/servicebus/perf/ReceiveAndDeleteMessageTest.java

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.messaging.servicebus.perf;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.microsoft.azure.messaging.servicebus.perf.core.ServiceBusStressOptions;
+import com.microsoft.azure.messaging.servicebus.perf.core.ServiceTest;
+import com.microsoft.azure.servicebus.IMessage;
+import com.microsoft.azure.servicebus.Message;
+import com.microsoft.azure.servicebus.ReceiveMode;
+import reactor.core.publisher.Mono;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Performance test.
+ */
+public class ReceiveAndDeleteMessageTest extends ServiceTest<ServiceBusStressOptions> {
+    private final ClientLogger logger = new ClientLogger(ReceiveAndDeleteMessageTest.class);
+    private final ServiceBusStressOptions options;
+
+    /**
+     * Creates test object
+     * @param options to set performance test options.
+     */
+    public ReceiveAndDeleteMessageTest(ServiceBusStressOptions options) {
+        super(options, ReceiveMode.RECEIVEANDDELETE);
+        this.options = options;
+    }
+
+    @Override
+    public Mono<Void> globalCleanupAsync() {
+        return Mono.fromFuture(sender.closeAsync())
+            .then(Mono.fromFuture(receiver.closeAsync()))
+            .then(super.globalCleanupAsync());
+    }
+
+    private Mono<Void> sendMessage() {
+        return Mono.defer(() -> {
+            String messageId = UUID.randomUUID().toString();
+            Message message = new Message(CONTENTS);
+            message.setMessageId(messageId);
+            int total = options.getParallel() * options.getMessagesToSend() * TOTAL_MESSAGE_MULTIPLIER;
+            List<Message> messages = IntStream.range(0, total)
```

Comment:
I'm not sure why you need this? You're just returning the message over and over again. You can do a foreach loop and add them rather than "mapping to an object". How do you discern between different messages?